tests/frontend/test_BoardGUI.py

```diff
@@ -0,0 +1,73 @@
+import unittest
+from unittest.mock import Mock, patch
+import pygame as pg
+from backend.GameManager import GameManager 
+from frontend.BoardGUI import BoardGUI 
+
+
+class TestBoardGUI(unittest.TestCase):
+    def setUp(self):
+       
+        self.mock_game = Mock()
+        self.mock_game.screen = Mock()
+        
+        pg.image.load = Mock()
+        pg.transform.smoothscale = Mock(return_value=Mock()) 
+       
+        self.board_size = 8  
+        self.win_size = 640  
+        self.total_pieces = 16
+        self.board_gui = BoardGUI(self.mock_game, self.win_size, self.board_size, self.total_pieces)
+
+    def tearDown(self):
+        # Auto cleanup done by unittest
+        pass
+
+    def test_initialization(self):
+        self.assertEqual(self.board_gui.cell_size, self.win_size // self.board_size)
+        self.assertIsInstance(self.board_gui.backend_game, GameManager)
+        self.assertEqual(self.board_gui.count, 0)
+
```

Comment:
Since we also initialize the board_image, black_piece_image and white_piece_image we should write some assertions for those here to reach higher code coverage.